static/extensions/DogeisCut/BeepBoxPlayer.js

```diff
@@ -0,0 +1,505 @@
<this.pitchChannelCount;e++)t.push(X[this.channels[e].octave]);for(let e=0;e<this.getChannelCount();e++)for(let s=0;s<this.channels[e].instruments.length;s++){const i=this.channels[e].instruments[s];if(t.push(84,X[i.type]),t.push(118,X[i.volume+u.volumeRange/2>>6],X[i.volume+u.volumeRange/2&63]),t.push(117,X[i.preset>>6],X[63&i.preset]),t.push(102),t.push(X[+i.eqFilterType]),i.eqFilterType)t.push(X[i.eqFilterSimpleCut]),t.push(X[i.eqFilterSimplePeak]);else{if(null==i.eqFilter)t.push(X[0]),console.log("Null EQ filter settings detected in toBase64String for channelIndex "+e+", instrumentIndex "+s);else{t.push(X[i.eqFilter.controlPointCount]);for(let e=0;e<i.eqFilter.controlPointCount;e++){const s=i.eqFilter.controlPoints[e];t.push(X[s.type],X[Math.round(s.freq)],X[Math.round(s.gain)])}}let n=0;for(let e=0;e<u.filterMorphCount-1;e++)n|=+(null!=i.eqSubFilters[e+1])<<e;t.push(X[n>>6],X[63&n]);for(let e=0;e<u.filterMorphCount-1;e++)if(n&1<<e){t.push(X[i.eqSubFilters[e+1].controlPointCount]);for(let s=0;s<i.eqSubFilters[e+1].controlPointCount;s++){const n=i.eqSubFilters[e+1].controlPoints[s];t.push(X[n.type],X[Math.round(n.freq)],X[Math.round(n.gain)])}}}if(t.push(113,X[i.effects>>6],X[63&i.effects]),P(i.effects))if(t.push(X[+i.noteFilterType]),i.noteFilterType)t.push(X[i.noteFilterSimpleCut]),t.push(X[i.noteFilterSimplePeak]);else{if(null==i.noteFilter)t.push(X[0]),console.log("Null note filter settings detected in toBase64String for channelIndex "+e+", instrumentIndex "+s);else{t.push(X[i.noteFilter.controlPointCount]);for(let e=0;e<i.noteFilter.controlPointCount;e++){const s=i.noteFilter.controlPoints[e];t.push(X[s.type],X[Math.round(s.freq)],X[Math.round(s.gain)])}}let n=0;for(let e=0;e<u.filterMorphCount-1;e++)n|=+(null!=i.noteSubFilters[e+1])<<e;t.push(X[n>>6],X[63&n]);for(let e=0;e<u.filterMorphCount-1;e++)if(n&1<<e){t.push(X[i.noteSubFilters[e+1].controlPointCount]);for(let s=0;s<i.noteSubFilters[e+1].controlPointCount;s++){const n=i.noteSubFilters[e+1].controlPoints[s];t.push(X[n.type],X[Math.round(n.freq)],X[Math.round(n.gain)])}}}if(M(i.effects)&&t.push(X[i.transition]),w(i.effects)&&(t.push(X[i.chord]),i.chord==u.chords.dictionary.arpeggio.index&&(t.push(X[i.arpeggioSpeed]),t.push(X[+i.fastTwoNoteArp]))),x(i.effects)&&t.push(X[i.pitchShift]),k(i.effects)&&t.push(X[i.detune-u.detuneMin>>6],X[i.detune-u.detuneMin&63]),F(i.effects)&&(t.push(X[i.vibrato]),i.vibrato==u.vibratos.length&&(t.push(X[Math.round(25*i.vibratoDepth)]),t.push(X[i.vibratoSpeed]),t.push(X[Math.round(i.vibratoDelay)]),t.push(X[i.vibratoType]))),I(i.effects)&&(t.push(X[i.distortion]),t.push(X[+i.aliases])),q(i.effects)&&t.push(X[i.bitcrusherFreq],X[i.bitcrusherQuantization]),T(i.effects)&&(t.push(X[i.pan>>6],X[63&i.pan]),t.push(X[i.panDelay])),D(i.effects)&&t.push(X[i.chorus]),O(i.effects)&&t.push(X[i.echoSustain],X[i.echoDelay]),R(i.effects)&&t.push(X[i.reverb]),4!=i.type&&(t.push(100,X[i.fadeIn],X[i.fadeOut]),t.push(X[+i.clicklessTransition])),5==i.type||7==i.type){t.push(7
```

Comment:
You shouldnt be using Timeouts in this way to check if playing. You should use `util.yield();`

See:
https://github.com/PenguinMod/PenguinMod-Vm/blob/ccf8c50db29c28a33a5f1189bff87f35ede6d12e/src/blocks/scratch3_control.js#L161